chasm/lib/callback/component.go

```diff
@@ -106,19 +102,24 @@ func (c *Callback) loadInvocationArgs(
 	}, nil
 }
 
+type saveResultInput struct {
+	result invocationResult
+	config *Config
+}
+
 func (c *Callback) saveResult(
 	ctx chasm.MutableContext,
-	result invocationResult,
+	input saveResultInput,
 ) (chasm.NoValue, error) {
-	switch r := result.(type) {
+	switch r := input.result.(type) {
 	case invocationResultOK:
 		err := TransitionSucceeded.Apply(ctx, c, EventSucceeded{Time: ctx.Now(c)})
 		return nil, err
 	case invocationResultRetry:
 		err := TransitionAttemptFailed.Apply(ctx, c, EventAttemptFailed{
 			Time:        ctx.Now(c),
 			Err:         r.err,
-			RetryPolicy: r.retryPolicy,
+			RetryPolicy: input.config.RetryPolicy(),
```

Comment:
Would you pass in just the retry policy here instead of the whole config just so we minimize the dependencies and make it easier to unit test?